app/workers/organizations/delete_worker.rb

```diff
@@ -0,0 +1,44 @@
+module Organizations
+  class DeleteWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :high_priority, retry: 10
+
+    # operator_id - the user who deletes the organization
+    def perform(organization_id, operator_id)
+      org = Organization.find_by(id: organization_id)
+      return unless org
+
+      user = User.find_by(id: operator_id)
+      return unless user
```

Comment:
Non-blocking question:

If a user asked to destroy an organization, then was destroyed (by deleting their account?) we would not want to destroy the organization? Are we guarding against forged user ids, or is a valid request from a past user no longer valid?

It looks like we have the user here for notification and cache busting, but that might be a secondary concern compared to the "please delete this organization" command being sent.

Or is the assumption that "since this user is no longer present, they're no longer an org-admin and don't have the power to remove the org"?

Reply:
Good point!

Technically, only the organization admin can destroy the user: https://github.com/forem/forem/blob/3dc8a386d2ea0b1ae957e6ec5a2cb276f7a577ad/app/policies/organization_policy.rb#L11

and we check authorization at the beginning 

https://github.com/forem/forem/blob/f66bc0de24dd9aaa7f3840398d485fcd00eac942/app/controllers/organizations_controller.rb#L53

If the user deletes themselves before the job is picked up by the queue, technically this will exit and not delete the org.

Not sure what we should do, maybe it's worth to raise a "user not found" error in this case?

Reply:
@djuber thanks for raising this question, now I think that in this case we still want to delete the org.